controllers/datadogagent/feature/kubernetesstatecore/feature.go

```diff
@@ -0,0 +1,174 @@
+				f.clusterChecksEnabled = true
+				f.rbacSuffix = common.CheckRunnersSuffix
+				f.serviceAccountName = v1alpha1.GetClusterChecksRunnerServiceAccount(dda)
+			}
+		} else {
+			f.serviceAccountName = v1alpha1.GetClusterAgentServiceAccount(dda)
+		}
+
+		if dda.Spec.Features.KubeStateMetricsCore.Conf != nil {
+			f.customConfig = v1alpha1.ConvertCustomConfig(dda.Spec.Features.KubeStateMetricsCore.Conf)
+		}
+
+		f.configConfigMapName = apicommonv1.GetConfName(dda, f.customConfig, apicommon.DefaultKubeStateMetricsCoreConf)
+	}
+
+	return f.enable
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *ksmFeature) ManageDependencies(managers feature.ResourcesManagers) error {
+	// Manage the Check Configuration in a configmap
+	configCM, err := f.buildKSMCoreConfigMap()
+	if err != nil {
+		return err
+	}
+	if configCM != nil {
+		managers.Store().AddOrUpdate(kubernetes.ConfigMapKind, configCM)
+	}
+
+	// Manager RBAC permission
+	rbacName := GetKubeStateMetricsRBACResourceName(f.owner, f.rbacSuffix)
+
+	return managers.RBACManager().AddClusterPolicyRules("", rbacName, f.serviceAccountName, getRBACPolicyRules())
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *ksmFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	// Manage KSM config in confimap
+	vol, volMount := volume.GetCustomConfigSpecVolumes(
+		f.customConfig,
+		apicommon.KubeStateMetricCoreVolumeName,
+		f.configConfigMapName,
+		ksmCoreCheckFolderName,
+	)
+
+	managers.Volume().AddVolumeToContainer(&vol, &volMount, common.ClusterAgentContainerName)
+
+	managers.EnvVar().AddEnvVar(&corev1.EnvVar{
+		Name:  apicommon.DDKubeStateMetricsCoreEnabled,
+		Value: "true",
+	})
+
+	managers.EnvVar().AddEnvVar(&corev1.EnvVar{
+		Name:  apicommon.DDKubeStateMetricsCoreConfigMap,
+		Value: f.configConfigMapName,
+	})
+
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *ksmFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	// Remove ksm v1 conf if the cluster checks are enabled and the ksm core is enabled
+	ignoreAutoConf := &corev1.EnvVar{
+		Name:  apicommon.DDIgnoreAutoConf,
+		Value: "kubernetes_state",
+	}
+
+	return managers.EnvVar().AddEnvVarToContainerWithMergeFunc(common.AgentContainerName, ignoreAutoConf, merger.AppendToValueEnvVarMergeFunction)
+}
+
+// ManageClusterCheckRunnerAgent allows a feature to configure the ClusterCheckRunnerAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *ksmFeature) ManageClusterCheckRunnerAgent(managers feature.PodTemplateManagers) error {
```

Comment:
another nit, but so i don't forget (for naming consistency)
```suggestion
func (f *ksmFeature) ManageClusterChecksRunner(managers feature.PodTemplateManagers) error {
```